sound/soc/sof/intel/hda-dsp.c

```diff
@@ -393,6 +393,10 @@ static int hda_dsp_update_d0i3c_register(struct snd_sof_dev *sdev, u8 value)
 static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 				const struct sof_dsp_power_state *target_state)
 {
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
```

Comment:
commit message: as discussed I think it makes sense to clarify that since the HDaudio legacy driver has no concept and support for D0ix, this is an SOF-only patch - not a change in sound/hda/hdac.